common/parallel/TreeCrawler.go

```diff
@@ -355,7 +358,10 @@ func (c *crawler) processOneDirectoryWithAutoPacer(ctx context.Context, workerIn
 	if c.isSync && c.isSource {
 		if _, ok := toExamine.(string); ok {
 			if _, ok := c.root.(string); ok {
-				c.tqueue <- common.RelativePath(toExamine.(string), c.root.(string))
+				c.output <- CrawlResult{
+					item:            common.RelativePath(toExamine.(string), c.root.(string)),
+					enqueueToTqueue: true,
+				}
```

Comment:
add a comment explaining what we are doing here.

//
// Add a special CrawlResult telling caller to enqueue this directory to tqueue for processing by the target traverser.
// Note that we don't enqueue it here but instead ask the caller to enqueue, to ensure that we add a directory to tqueue only after all the directory children have been added to the objectIndexer map.
//